File: engineer_man/csharp/easy/reverse_a_string.cs
using System;
using System.Linq;

class MainClass {
    static void Main(string[] args) {
        /*
        Simple string
            value1 cat
            output tac

        Complex string
            value1 sd#$)(&*09&M0
            output 0M&90*&()$#ds
        */
        string value1 = args[0];

        var chars = value1.ToArray();
        Array.Reverse(chars);
        Console.WriteLine(new string(chars));
    }
}

File: engineer_man/c/easy/reverse_a_string.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char **argv) {
    /*
    Simple string
        value1 cat
        output tac

    Complex string
        value1 sd#$)(&*09&M0
        output 0M&90*&()$#ds
    */
    char value1[128];
    strcpy(value1, argv[1]);

    int endIdx;
    for (int i = 0; value1[i] != '\0'; i++)
    {
        endIdx = i;
    }

    char final[128];
    for (int i = endIdx;  i >= 0; i--)
    {
        final[endIdx-i] = value1[i];
    }
    final[endIdx+1] = '\0';

    printf("%s", final);
}